server/etcdserver/api/etcdhttp/health.go

```diff
@@ -91,6 +98,31 @@ func NewHealthHandler(lg *zap.Logger, hfunc func(excludedAlarms AlarmSet, Serial
 	}
 }
 
+// newHealthHandler generates a http HandlerFunc for a health check function hfunc.
+func newHealthHandler(path string, lg *zap.Logger, hfunc func(*http.Request) (Health, error)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			w.Header().Set("Allow", http.MethodGet)
+			http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+			lg.Warn("/health error", zap.Int("status-code", http.StatusMethodNotAllowed))
+			return
+		}
+		h, err := hfunc(r)
+		if err != nil {
+			panic(err)
```

Comment:
From this place it's hard to know why error here should cause panic. I would recommend removing error from hfunc return, and panic on line 284 directly.

Reply:
done